connector/countconnector/counter.go

```diff
@@ -48,10 +50,16 @@ type attrCounter struct {
 
 func (c *counter[K]) update(ctx context.Context, attrs pcommon.Map, tCtx K) error {
 	var errors error
+
 	for name, md := range c.metricDefs {
 		countAttrs := pcommon.NewMap()
 		for _, attr := range md.attrs {
-			if attrVal, ok := attrs.Get(attr.Key); ok {
+			attrVal, ok := attrs.Get(attr.Key)
+			if !ok && attr.MaxDepth > 1 {
```

Comment:
I think we need to tie this condition into the if/else below. Otherwise, `!ok` could result in the default value being applied.